microsoft/iot-cardboard-js
# [React-select for ResourcePicker]: Component refactors

### Summary of changes 🔍 
- Replaced fluent combobox with [react-select](https://react-select.com/components) component in `ResourcePicker ` component to get nice features like search&filtered list of options and made necessary changes in parent component and data interaction flow
- Since react-select's perf is really bad with long lists, as suggested [here ](https://github.com/JedWatson/react-select/issues/3128) I added [react-window](https://www.npmjs.com/package/react-window) package to render only visible part of long list with thousands of options in dropdown (called virtualized list - I used `VirtualizedComboBox` from fluent before to achieve this) as I pull all the resources in single call in `AzureManagementAdapter` (**not** making async calls as user scrolls down by loading more options, if this is requested I need to change AzureManagementAdapter `getResourcesByPermission` method)

![image](https://user-images.githubusercontent.com/45217314/202818543-6944bbb0-d69c-427f-b794-178693ac3371.png)
![image](https://user-images.githubusercontent.com/45217314/202818691-fae80935-c4a3-4df4-b53b-6e14be5d38b2.png)
![image](https://user-images.githubusercontent.com/45217314/202818720-f995394f-7f7c-4e9f-8301-d4a09657a2cf.png)
![image](https://user-images.githubusercontent.com/45217314/202818924-fbee6868-615e-4767-9672-34e9d58d143c.png)
![image](https://user-images.githubusercontent.com/45217314/202818949-5352a14a-718e-4435-ba7d-89430021dcdf.png)
![image](https://user-images.githubusercontent.com/45217314/202819004-617b6d44-8975-43fc-bb51-e563d9d91848.png)
![image](https://user-images.githubusercontent.com/45217314/202819052-ab445c7e-137b-476a-8dd0-d3249581de7e.png)


### Testing 🧪
Test it with either standalone ResourcePicker stories or by opening EnvironmentPicker in ADT3DScenePage. Try:
- search&auto complete, menu list interactions
- clear input
- create additional options with "No match found, create ..." option
- check the dependency between storage account and container pickers (whenever you change, clear a storage account - container picker should fetc/update its options accordingly)

### Checklist ✔️
